### PR TITLE
Identification

### PR DESCRIPTION
**Pull request by Solodkih**

### CODE REVIEW

File: frontend/src/App/component/Menu/ButtonsMainMenu/index.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
+import React from 'react';
```
Comment (porlov) [inline frontend/src/App/component/Menu/ButtonsMainMenu/index.js:1]: lets have the following structure:

`Menu/buttons/LoginRegistrationButtons.js`
`Menu/buttons/ProfileLink.js`
`Menu/buttons/MainMenuButtons.js`
`Menu/buttons/index.js` - contains **only** one line - export `MainMenuButtons`
`Menu/navigation/NavigationBar.js`
`Menu/navigation/NavigationContainer.js`
`Menu/navigation/index.js` - exports `NavigationContainer`
`Menu/index.js` - exports `NavigationContainer` from `Menu/navigation/index.js`

File: frontend/src/App/component/Menu/index.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+import PropTypes from 'prop-types';
+import Registration from '../User/components/Registration';
+import Login from '../User/components/Login';
+import ButtonsMainMenu from './ButtonsMainMenu';
+
+import * as actUI from '../../action/actionsUI';
+import * as actUser from '../User/actions/actionsUser';
+
+
+function NavigationContainer(props) {
```
Comment (porlov) [inline frontend/src/App/component/Menu/index.js:18]: These components are duplicates from file `Navigation.js`

File: frontend/src/App/component/User/actions/axios.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  const response = await axios.post(urls.login, { email, password });
+  return response;
+}
+
+export async function getMyId() {
+  const response = await axios.get(urls.me);
+  return response.data;
+}
+
+export async function getUs(id) {
```
Comment (porlov) [inline frontend/src/App/component/User/actions/axios.js:33]: rename to `getProfile`

File: frontend/src/App/component/User/actions/axios.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  const response = await axios.get(`${urls.detail}${id}`);
+  const newUser = {
```
Comment (porlov) [inline frontend/src/App/component/User/actions/axios.js:35]: rename to `profile`

File: frontend/src/App/component/User/actions/actionsUserTests/actionsUserGetMe.test.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+    getMyId.mockResolvedValue(user);
+    getUs.mockResolvedValue(user);
+    setProfile.mockReturnValue(action);
+    const dispatch = jest.fn();
+
+    await getMe()(dispatch);
+    expect(getMyId).toBeCalled();
+    expect(getUs).toHaveBeenCalledWith(user.id);
+    expect(dispatch).toHaveBeenCalled();
```
Comment (porlov) [inline frontend/src/App/component/User/actions/actionsUserTests/actionsUserGetMe.test.js:36]: we should test that `setProfile` is called instead. For this we should dispatch action in the store.

File: frontend/src/App/component/User/actions/actionsUserTests/actionsUserLogin.test.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    getMe.mockClear();
+  });
+
+  it('when login is successful', async () => {
+    const user = { email: 'email', password: 'password' };
+    const responce = { status: 'good' };
+    loginUser.mockResolvedValue(responce);
+    const dispatch = jest.fn();
+    await login(user)(dispatch);
+    expect(dispatch).toHaveBeenCalled();
```
Comment (porlov) [inline frontend/src/App/component/User/actions/actionsUserTests/actionsUserLogin.test.js:20]: the same for this one

File: frontend/src/App/component/User/actions/actionsUserTests/actionsUserRegistration.test.js [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+  it('when registartion is successful', async () => {
+    const user = { email: 'test@test.ru', password: '123456' };
+    registerUser.mockResolvedValue(user);
+    const dispatch = jest.fn();
+    await registrationUser(user)(dispatch);
+    expect(dispatch).toHaveBeenCalled();
+    expect(registerUser).toHaveBeenCalledWith(user);
+  });
+
+  it('when registartion is fail', async () => {
```
Comment (porlov) [inline frontend/src/App/component/User/actions/actionsUserTests/actionsUserRegistration.test.js:21]: typo. Please check all occurrences.

File: frontend/src/App/component/User/actions/axios.js [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+    const response = await axios.post(urls.list, newUser);
+    return response.data;
+  } catch (result) {
+    const errors = getErrors(result.response.data);
+    throw errors;
+  }
+}
+
+export async function loginUser({ email, password }) {
+  const response = await axios.post(urls.login, { email, password });
```
Comment (porlov) [inline frontend/src/App/component/User/actions/axios.js:24]: Where is the errors under login? :slightly_smiling_face:

File: frontend/src/App/component/User/components/Login/LoginFormPresent.js [hunk 9]
```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import Form from 'react-bootstrap/Form';
+import PropTypes from 'prop-types';
+
+function LoginFormPresent({ handleChangeEmail, handleChangePassword }) {
```
Comment (porlov) [inline frontend/src/App/component/User/components/Login/LoginFormPresent.js:5]: Lets name all present components without prefix: `LoginForm`

File: frontend/src/App/index.js [hunk 10]
```diff
@@ -1,19 +1,53 @@
...
 
 
-function App() {
+import { BrowserRouter, Switch } from 'react-router-dom';
+import * as actUser from './component/User/actions/actionsUser';
+import Navigation from './component/Menu/Navigation';
+import Routing from './component/Routing';
+
+
+function AppContain({ getMe, profile }) {
```
Comment (porlov) [inline frontend/src/App/index.js:12]: `AppContainer`?

File: frontend/src/App/action/actionsUI.js [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+  };
+}
+
+export function switchOffLoginForm() {
+  return {
+    type: HIDE_LOGIN_FORM,
+  };
+}
+
+export function clearAll() {
```
[the remark this answers is not in the document]
Comment (porlov) [reply frontend/src/App/action/actionsUI.js:32]: https://stackoverflow.com/questions/35622588/how-to-reset-the-state-of-a-redux-store

File: frontend/src/App/component/User/actions/axios.js (hunk 3 above)
Comment (Solodkih) [reply frontend/src/App/component/User/actions/axios.js:33]: I can’t call him that, it's just to get a user

File: frontend/src/App/component/User/actions/axios.js (hunk 4 above)
Comment (Solodkih) [reply frontend/src/App/component/User/actions/axios.js:35]: I can’t call him that, it's just to get a user

File: frontend/src/App/component/User/actions/axios.js (hunk 8 above)
Comment (Solodkih) [reply frontend/src/App/component/User/actions/axios.js:24]: In component

File: frontend/src/App/component/User/components/Registration/index.js [hunk 12]
```diff
@@ -0,0 +1,75 @@
+import React, { useState } from 'react';
```
Comment (porlov) [inline frontend/src/App/component/User/components/Registration/index.js:1]: Please move to separate file too and make index.js simpler

File: frontend/src/App/component/User/components/Login/index.js [hunk 13]
```diff
@@ -0,0 +1,90 @@
+import React, { useState } from 'react';
```
Comment (porlov) [inline frontend/src/App/component/User/components/Login/index.js:1]: Please move to separate file too and make index.js simpler

File: frontend/src/App/component/User/actions/axios.js (hunk 4 above)
Comment (porlov) [reply frontend/src/App/component/User/actions/axios.js:35]: newUser -> userDetail

File: frontend/src/constants.js [hunk 14]
```diff
@@ -0,0 +1,12 @@
+export const userRole = {
+  PAINTER: 'PA',
+  BUYER: 'BU',
+};
+
+export const urls = {
+  list: '/api/v1/users/',
+  login: '/api/v1/users/login',
```
[the remark this answers is not in the document]
Comment (Solodkih) [reply frontend/src/constants.js:8]: We discussed this. This is necessary when going to "/ user" and to "user / login" you need to go through the slash, Django himself can set "/", but the documentation says that during POST data can be lost

**Pull request merged**